getting_started/scripting/gdscript/static_typing.rst

```diff
@@ -0,0 +1,407 @@
+        pass
+
+And the same callback, with type hints:
+
+::
+
+    func _on_area_entered(area : CollisionObject2D) -> void:
+        pass
+
+You’re free to replace, e.g. the ``PhysicsBody2D``, with your own type,
+to cast parameters automatically:
+
+::
+
+    func _on_area_entered(bullet : Bullet) -> void:
+        if not bullet:
+            return
+        take_damage(bullet.damage)
+
+The ``bullet`` variable could hold any ``CollisionObject2D`` here, but
+we make sure it is our ``Bullet``, a node we created for our project. If
+it’s anything else, like an ``Area2D``, or any node that doesn’t extend
+``Bullet``, the ``bullet`` variable will be ``null``.
+
+Warning system
+--------------
+
+The warning system complements typed GDscript. It’s here to help you
+avoid mistakes that are hard to spot during development, and that may
+lead to runtime errors.
+
+You can configure warnings in the Project Settings under a new section
+called ``GDscript``:
+
+.. figure:: ./img/typed_gdscript_warning_system_settings.png
+   :alt: warning system project settings
+
+   warning system project settings
+
+You can find a list of warnings for the active GDscript file in the
+script editor’s status bar. The example below has 3 warnings:
+
+.. figure:: ./img/typed_gdscript_warning_example.png
+   :alt: warning system example
+
+   warning system example
+
+To ignore specific warnings in one file, insert a special comment of the
+form ``#warning-ignore:warning-id``, or click on the ignore link to the
+right of the warning’s description. Godot will add a comment above the
+corresponding line and the code won’t trigger the corresponding warning
+anymore:
+
+.. figure:: ./img/typed_gdscript_warning_system_ignore.png
+   :alt: warning system ignore example
+
+   warning system ignore example
+
+Warnings won’t prevent the game from running, but you can turn them into
+errors if you’d like. This way your game won’t compile unless you fix
+all warnings. Head to ``GDscript`` section of the Project Settings to
+turn on this option. Here’s the same file as the previous example with
+warnings as errors turned on:
+
+.. figure:: ./img/typed_gdscript_warning_system_errors.png
+   :alt: warnings as errors
+
+   warnings as errors
+
+Cases where you can’t specify types
+-----------------------------------
+
+To wrap up this introduction, let’s cover a few cases where you can’t
+use type hints. All the examples below **will trigger errors**.
+
+You can’t use Enums as types:
+
+::
+
+    enum MoveDirection { UP, DOWN, LEFT, RIGHT }
+    var current_direction : MoveDirection
+
+You can’t specify the type of individual members in an array. This will
+give you an error:
+
+::
+
+    var enemies : Array = [$Goblin : Enemy, $Zombie : Enemy]
+
+You can’t force the assignment of types in a ``for`` loop as each
+element the ``for`` keyword loops already has a different type. So you
+**cannot** write:
+
+::
+
+    var names ['John', 'Marta', 'Samantha', 'Jimmy']
```

Comment:
I haven't tested the code myself, but I imagine you meant to add an `=` between `names` and the array?